extensions/ffmpeg/src/main/java/com/google/android/exoplayer2/ext/ffmpeg/FfmpegDecoder.java

```diff
@@ -87,7 +88,17 @@ public FfmpegDecoderException decode(DecoderInputBuffer inputBuffer,
     }
     if (!hasOutputFormat) {
       channelCount = ffmpegGetChannelCount(nativeContext);
-      sampleRate = ffmpegGetSampleRate(nativeContext);
+
+      // Unlike the other decoders, ALAC does not set the sample rate, so we must get it ourselves.
+      // See https://trac.ffmpeg.org/ticket/6096
+      if ("alac".equals(codecName)) {
+        ParsableByteArray parsableExtraData = new ParsableByteArray(extraData);
+        parsableExtraData.setPosition(extraData.length - 4);
+        sampleRate = parsableExtraData.readUnsignedIntToInt();
+      } else {
+        sampleRate = ffmpegGetSampleRate(nativeContext);
```

Comment:
One final question. Does this return 0 currently? If so, and given it looks like the FFMPEG issue is going to be fixed, I was wondering whether we could structure the code so the workaround is only performed if using a version of FFMPEG before the fix lands, like (warning: untested, so please test this!):

```
sampleRate = ffmpegGetSampleRate(nativeContext);
if (sampleRate == 0 && "alac".equals(codecName)) {
  // ALAC decoder did not set the sample rate in earlier versions of FFMPEG. See:
  // See https://trac.ffmpeg.org/ticket/6096
  ParsableByteArray parsableExtraData = new ParsableByteArray(extraData);
  parsableExtraData.setPosition(extraData.length - 4);
  sampleRate = parsableExtraData.readUnsignedIntToInt();
}
```